launch_testing/launch_testing/__init__.py

```diff
@@ -35,6 +35,7 @@ def __init__(
         self.name = name
         self.launch_descriptor = launch_descriptor
         self.expected_lines = expected_lines
+        self.expected_output = b'\n'.join(self.expected_lines)
```

Comment:
Yes, as I said below, I'm not claiming to cover every situation in this pull request.

I still think this is adequate for existing use cases, since the previous functionality of launch_testing only matched a single-line regex to the output.

It creates room for error in that miscellaneous output could be printed to the screen and the test would still pass in tests that expect a regex.

However, in the places where this package is used (examples and demos) the expectation is generally a line that is printed when some set of conditions is met. e.g. a message has been received by a subscriber.

So if the miscellaneous output printed to the screen is due to a fatal error that causes the tested process to crash before meeting the required condition, then the test will fail.